tree/dataframe/inc/ROOT/RDF/RDatasetSpec.hxx

```diff
@@ -13,44 +13,100 @@
 
 #include <string>
 #include <vector>
+#include <limits>
+#include <stdexcept> // std::logic_error
+#include <algorithm> // std::transform
 
 #include <RtypesCore.h>
+#include <ROOT/InternalTreeUtils.hxx>
 
 namespace ROOT {
 
 namespace RDF {
 
 struct RDatasetSpec {
-   std::string fDatasetName{}; ///< The name of the dataset to process.
-   /**
-    * A list of file names.
-    * They can contain the globbing characters supported by TChain. See TChain::Add for more information.
-    */
-   std::vector<std::string> fFileNameGlobs{};
-   ULong64_t fStartEntry{};                    ///< The entry where the dataset processing should start (inclusive).
-   ULong64_t fEndEntry{};                      ///< The entry where the dataset processing should end (exclusive).
-   std::vector<std::string> fDefaultColumns{}; ///< A list of column names to process in the dataset.
+
+   struct REntryRange {
+      ULong64_t fStartEntry{0};
+      ULong64_t fEndEntry{std::numeric_limits<ULong64_t>::max()};
+      REntryRange() {}
+      REntryRange(ULong64_t endEntry) : fEndEntry(endEntry) {}
+      REntryRange(ULong64_t startEntry, ULong64_t endEntry)
+         : fStartEntry(startEntry),
+           fEndEntry(endEntry >= startEntry
+                        ? endEntry
+                        : throw std::logic_error("RDatasetSpec: fStartEntry cannot be larger than fEndEntry."))
+      {
+      }
+   };
+
    /**
     * A list of names of trees.
     * This list should go in lockstep with fFileNameGlobs, only in case this dataset is a TChain where each file
     * contains its own tree with a different name from the global name of the dataset.
     */
-   std::vector<std::string> fSubTreeNames{};
+   std::vector<std::string> fTreeNames{};
+
+   /**
+    * A list of file names.
+    * They can contain the globbing characters supported by TChain. See TChain::Add for more information.
+    */
+   std::vector<std::string> fFileNameGlobs{};
```

Comment:
Exactly like `RFriendInfo`, also `RDataSetSpec` should have three data members: one string with the name of the tree, one vector with the filenames and one vector with the (optional) names of trees in the files. The constructors should behave similarly to the overloads of `AddFriend` when filling this data members

Reply:
A question in the process of implementing. This optional argument seems applicable only for the third overload, of different tree names. Right?

Reply:
Alternative is to call the chain the same as the tree whenever I have only 1 tree name. And have no chain name, when I have different tree names in the cahin. this version is closer to our previous conclusion.